packages/console/src/HeapUsage.tsx

```diff
@@ -41,9 +41,16 @@ function HeapUsage({
     usages: [],
   });
 
+  const isMountedRef = useIsMountedRef();
+
   const setUsageUpdateInterval = useCallback(async () => {
     try {
       const newUsage = await connection.getWorkerHeapInfo();
+
+      if (!isMountedRef.current) {
+        return;
+      }
```

Comment:
Technically this may be incorrect. E.g.
1. First pass, `monitorDuration` is `5000`, and a call to `connection.getWorkerHeapInfo()` is triggered. It is async and hasn't returned yet
2. `monitorDuration` changes to `10000`, triggering another call to `connection.getWorkerHeapInfo()`. This call returns before the call from step 1
3. Call from step 1 returns, now the data is out of date.

In practice that could never happen in this case, since the first call to `connection.getWorkerHeapInfo()` should always return before a subsequent call, but in another async situation, using the `isMountedRef` may not handle all cases.

I typically go with the `isCancelled` pattern, as that handles cases where the dependencies change, not just unmounting. An example from the React docs (looks like they name it `ignore` instead of `isCancelled` now): https://react.dev/reference/react/useEffect#fetching-data-with-effects

Reply:
I think `monitorDuration` is how long it holds onto the history and doesn't impact the interval (unless I am missing something)

I think your appoint does apply to the interval though `isOpen ? hoverUpdateInterval : defaultUpdateInterval`. So ideally we should be cancelling if either the component unmounts `or` if the interval changes?